src/components/Schedule/SearchBar.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { CloseIcon, SearchIcon } from "@chakra-ui/icons";
+import {
+  IconButton,
+  Input,
+  InputGroup,
+  InputRightElement,
+} from "@chakra-ui/react";
+import { Course } from "../../stores/schedule";
+
+interface SearchBarProps {
+  termData: Course[];
+  setScheduleData: (data: Course[]) => void;
+}
+
+export const SearchBar = (props: SearchBarProps) => {
+  const { termData, setScheduleData } = props;
+
+  const [searchInput, setSearchInput] = useState("");
+  const [filtered, setFiltered] = useState("");
+
+  const filterCourses = (clear?: boolean) => {
+    if (searchInput === "" || clear) {
+      setFiltered("");
+      return setScheduleData(termData);
+    }
+
+    //checks if every word of the search exists as an attribute for the course
+    const input = searchInput.toLowerCase().split(" ");
+    const filteredAppointments = termData.filter(
```

Comment:
Might want to rename this from `input` to `inputWords` or something more descriptive

Reply:
updated